scripts/files/tests/fs_s3_test.py

```diff
@@ -124,3 +124,20 @@ def test_exists_object_starting_with_not_exists() -> None:
     file_exists = exists("s3://testbucket/hello/another.fi")
 
     assert file_exists is False
+
+
+@mock_s3  # type: ignore
+def test_list_files_in_uri() -> None:
+    s3 = resource("s3", region_name=DEFAULT_REGION_NAME)
+    boto3_client = client("s3", region_name=DEFAULT_REGION_NAME)
+    s3.create_bucket(Bucket="testbucket")
+    boto3_client.put_object(Bucket="testbucket", Key="data/collection.json", Body=b"test content")
+    boto3_client.put_object(Bucket="testbucket", Key="data/image.tiff", Body=b"test content")
+    boto3_client.put_object(Bucket="testbucket", Key="data/image_meta.xml", Body=b"test content")
+
+    files = list_files_in_uri("s3://testbucket/data/", [".json", "_meta.xml"], boto3_client)
+
+    assert len(files) == 2
+    assert "data/collection.json" in files
+    assert "data/image_meta.xml" in files
+    assert "data/image.tiff" not in files
```

Comment:
Could we simply

```suggestion
    assert files == ["data/collection.json", "data/image_meta.xml"]
```

or, if the sequence is actually arbitrary,

```suggestion
    assert set(files) == {"data/collection.json", "data/image_meta.xml"}
```